### PR TITLE
Intro: Add Brier score

### LINKED ISSUE #310: Include Brier score in "Evaluating models" deck
From https://github.com/tidymodels/workshops/pull/305#discussion_r1687918056

>  It would be nice to include the Brier score since it's now one of the default metrics in tune, no?

### PR DESCRIPTION
**Pull request by hfrick**
closes #310 

It's used in the exercise to demonstrate overfitting, so I didn't add another exercise here.

### CODE REVIEW

File: slides/intro-04-evaluating-models.qmd [hunk 1]
```diff
@@ -259,6 +259,70 @@ augment(forested_fit, new_data = forested_train) %>%
 countdown::countdown(minutes = 5, id = "roc-curve")
 ```
 
+## Brier score
+
+What if we don't turn predicted probabilties into class predictions?
+
+. . .
+
+The Brier score is analogous to the mean squared error in regression models:
```
Comment (hfrick) [inline slides/intro-04-evaluating-models.qmd:268]: From https://yardstick.tidymodels.org/reference/brier_class.html#details

File: slides/intro-04-evaluating-models.qmd [hunk 2]
```diff
@@ -259,6 +259,70 @@ augment(forested_fit, new_data = forested_train) %>%
...
+```
+
+. . .
+
+Smaller values are better, for binary classification the "bad model threshold" is about 0.25.
+
+
+## Separation vs calibration
+
+The ROC captures separation, the Brier score captures calibration.
```
Comment (hfrick) [inline slides/intro-04-evaluating-models.qmd:288]: This is adapted from https://www.tidymodels.org/learn/statistics/survival-metrics/#disagreement-between-metrics

File: slides/intro-04-evaluating-models.qmd [hunk 3]
```diff
@@ -259,6 +259,70 @@ augment(forested_fit, new_data = forested_train) %>%
...
+## Brier score
+
+```{r brier}
+augment(forested_fit, new_data = forested_train) %>% 
+  brier_class(truth = forested, .pred_Yes) 
+```
+
+. . .
+
+Smaller values are better, for binary classification the "bad model threshold" is about 0.25.
```
Comment (hfrick) [inline slides/intro-04-evaluating-models.qmd:283]: Ref for `0.25`: https://www.tidymodels.org/learn/models/calibration/#a-naive-bayes-model

Comment (simonpcouch) [reply slides/intro-04-evaluating-models.qmd:283]: Just read that linked section--what's the explanation for why .25? Should we drop it in a `:::notes`?

File: slides/intro-04-evaluating-models.qmd [hunk 4]
```diff
@@ -259,6 +259,70 @@ augment(forested_fit, new_data = forested_train) %>%
 countdown::countdown(minutes = 5, id = "roc-curve")
 ```
 
+## Brier score
+
+What if we don't turn predicted probabilties into class predictions?
```
Comment (simonpcouch) [inline slides/intro-04-evaluating-models.qmd:264]:
```suggestion
What if we don't turn predicted probabilities into class predictions?
```

File: slides/intro-04-evaluating-models.qmd (hunk 2 above)
Comment (simonpcouch) [reply slides/intro-04-evaluating-models.qmd:288]: This is a super artful slide.

I'm 50/50 on whether this is the right approach, but: the current structure of the slide _seems to suggest_ that the LHS plot refers to the comment on separation and RHS on calibration:

Screenshot 2024-07-29 at 9 26 22 AM

Should we leave no ambiguity / room for doubt there by including "The ROC captures separation" in the first `:::columns` and "The Brier score captures calibration" in the second?

**Review by simonpcouch — APPROVED** [review]
Kickass. This is a great addition.🦑

File: slides/intro-04-evaluating-models.qmd (hunk 3 above)
Comment (hfrick) [reply slides/intro-04-evaluating-models.qmd:283]: That's a good question -- @topepo do you have some context for us here?

File: slides/intro-04-evaluating-models.qmd (hunk 2 above)
Comment (hfrick) [reply slides/intro-04-evaluating-models.qmd:288]: good point, I'll try it out!

Comment (hfrick) [reply slides/intro-04-evaluating-models.qmd:288]: Screenshot 2024-07-29 at 17 11 16

File: slides/intro-04-evaluating-models.qmd (hunk 3 above)
Comment (topepo) [reply slides/intro-04-evaluating-models.qmd:283]: I can add an annotation. For two classes (under certain conditions) the worst possible model predicts everything to be 1/2, so it is (1 - 1/2)^2

Comment (topepo) [reply slides/intro-04-evaluating-models.qmd:283]: Add to this PR or add another?

Comment (topepo) [reply slides/intro-04-evaluating-models.qmd:283]: Added it here

Comment (hfrick) [reply slides/intro-04-evaluating-models.qmd:283]: Nice one, thank you! 🙌

**Pull request merged**